### PR TITLE
Migrate crictl to docker-cli mapping to references

### PR DESCRIPTION
**Pull request by shannonxtreme**
Issue: #30976 

This PR migrates a portion of `crictl.md` that maps `docker-cli` commands to equivalent `crictl` commands to the References > Other Tools section and insta-deprecates the page. 

Also updates the `crictl.md` docs to mention default isn't `dockershim.sock` and to mention that default endpoints are deprecated. 

/sig docs
/sig node
/language en
/cc @sftim @SergeyKanzhelev

### CODE REVIEW

File: content/en/docs/tasks/debug-application-cluster/crictl.md [hunk 1]
```diff
@@ -41,13 +39,19 @@ of Kubernetes. Extract it and move it to a location on your system path, such as
 The `crictl` command has several subcommands and runtime flags. Use
 `crictl help` or `crictl <subcommand> help` for more details.
 
-`crictl` connects to `unix:///var/run/dockershim.sock` by default. For other
+`crictl` connects to `unix:///var/run/containerd.sock` by default. For other
```
Comment (afbjorklund) [inline content/en/docs/tasks/debug-application-cluster/crictl.md:42]: there is no default runtime, and "other runtimes", anymore. crictl will search a list of known runtimes, that's all

most likely you **don't** want to duplicate that list of endpoints here, but link to the crictl documentation instead ?

File: content/en/docs/tasks/debug-application-cluster/crictl.md [hunk 2]
```diff
@@ -56,8 +60,8 @@ To view or edit the current configuration, view or edit the contents of `/etc/cr
 
 ```shell
 cat /etc/crictl.yaml
-runtime-endpoint: unix:///var/run/dockershim.sock
-image-endpoint: unix:///var/run/dockershim.sock
+runtime-endpoint: unix:///var/run/containerd.sock
```
Comment (afbjorklund) [inline content/en/docs/tasks/debug-application-cluster/crictl.md:63]: this is wrong, containerd socket lives in `/run/containerd/containerd.sock`

**Review by afbjorklund — CHANGES_REQUESTED** [review]
More work is needed for crictl, and it probably should move to the main "container runtimes" ?

Hiding a required tool under "node debugging" seems a bit odd, and has been so for a while

File: content/en/docs/reference/tools/map-crictl-dockercli.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+This page provides a reference for mapping common docker cli commands to
+equivalent crictl commands.
+
+## Mapping from docker cli to crictl
+
+The exact versions for below mapping table are for docker cli v1.40 and crictl v1.19.0. Please note that the list is not exhaustive. For example, it doesn't include experimental commands of docker cli.
+
+{{< note >}}
+The output format of CRICTL is similar to Docker CLI, despite some missing columns for some CLI. Make sure to check output for the specific command if your script output parsing.
```
Comment (SergeyKanzhelev) [inline content/en/docs/reference/tools/map-crictl-dockercli.md:23]:
```suggestion
The output format of CRICTL is similar to Docker CLI, despite some missing columns for some CLI. Make sure to check output for the specific command if your command output is being parsed programmatically.
```

File: content/en/docs/tasks/debug-application-cluster/crictl.md (hunk 1 above)
Comment (shannonxtreme) [reply content/en/docs/tasks/debug-application-cluster/crictl.md]: I changed the wording to this and linked to the crictl docs for more information. 

You can set the endpoint for `crictl` by doing one of the following:

* Set the `--runtime-endpoint` and `--image-endpoint` flags.
* Set the `CONTAINER_RUNTIME_ENDPOINT` and `IMAGE_SERVICE_ENDPOINT` environment
  variables.
* Set the endpoint in the configuration file `--config=/etc/crictl.yaml`.

{{<note>}}
If you don't set an endpoint, `crictl` attempts to connect to a list of known
endpoints, which might result in an impact to performance.
{{</note>}}

File: content/en/docs/reference/tools/map-crictl-dockercli.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+---
+title: Mapping from dockercli to crictl
+content_type: reference
+---
+
+{{<caution>}}
+This page is deprecated and will be removed in Kubernetes 1.27.
+{{</caution>}}
```
Comment (sftim) [inline content/en/docs/reference/tools/map-crictl-dockercli.md:8]: Use a note callout here. Caution is for advice that you should heed if you don't want to put your cluster at risk (warning is for “never do this: _x_” notices).

File: content/en/docs/tasks/debug-application-cluster/crictl.md (hunk 2 above)
Comment (shannonxtreme) [reply content/en/docs/tasks/debug-application-cluster/crictl.md]: Led into this with a "for example, the configuration for the containerd container runtime would be similar to this:"

File: content/en/docs/reference/tools/map-crictl-dockercli.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+`crictl` is a command-line interface for {{<glossary_tooltip term_id="cri" text="CRI">}}-compatible container runtimes.
+You can use it to inspect and debug container runtimes and applications on a
+Kubernetes node. `crictl` and its source are hosted in the
+[cri-tools](https://github.com/kubernetes-sigs/cri-tools) repository.
+
+This page provides a reference for mapping common docker cli commands to
+equivalent crictl commands.
+
+## Mapping from docker cli to crictl
```
Comment (sftim) [inline content/en/docs/reference/tools/map-crictl-dockercli.md:18]:
```suggestion
## Mapping from docker CLI to crictl
```
We usually capitalize abbreviations.
It's OK to skip that as this is moving existing content.

File: content/en/docs/tasks/debug-application-cluster/crictl.md [hunk 6]
```diff
@@ -354,58 +359,4 @@ CONTAINER ID        IMAGE               CREATED              STATE
...
-`imagefsinfo` | Return image filesystem info
-`inspectp` | Display the status of one or more pods
-`port-forward` | Forward local port to a pod
-`pods` | List pods
-`runp` | Run a new pod
-`rmp` | Remove one or more pods
-`stopp` | Stop one or more running pods
-{{< /table >}}
\ No newline at end of file
+for more information.
```
Comment (sftim) [inline content/en/docs/tasks/debug-application-cluster/crictl.md:362]: Should we hyperlink to /docs/reference/tools/map-crictl-dockercli/ ?

File: content/en/docs/reference/tools/_index.md [hunk 7]
```diff
@@ -21,7 +21,14 @@ development and testing purposes.
...
 [`Dashboard`](/docs/tasks/access-application-cluster/web-ui-dashboard/), the web-based user interface of Kubernetes, allows you to deploy containerized applications
-to a Kubernetes cluster, troubleshoot them, and manage the cluster and its resources itself.
+to a Kubernetes cluster, troubleshoot them, and manage the cluster and its
+resources itself.
+
+## crictl
+
+[`crictl`](https://github.com/kubernetes-sigs/cri-tools) is a command-line
+interface for inspecting and debugging {{<glossary_tooltip term_id="cri" text="CRI">}}-compatible
+container runtimes.
```
Comment (sftim) [inline content/en/docs/reference/tools/_index.md:31]: Let's shift the list to alphabetical sort (could be a separate commit). We've already reached a point where we should be doing that.

File: content/en/docs/reference/tools/map-crictl-dockercli.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+This page is deprecated and will be removed in Kubernetes 1.27.
+{{</caution>}}
+
+`crictl` is a command-line interface for {{<glossary_tooltip term_id="cri" text="CRI">}}-compatible container runtimes.
+You can use it to inspect and debug container runtimes and applications on a
+Kubernetes node. `crictl` and its source are hosted in the
+[cri-tools](https://github.com/kubernetes-sigs/cri-tools) repository.
+
+This page provides a reference for mapping common docker cli commands to
+equivalent crictl commands.
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/reference/tools/map-crictl-dockercli.md:16]: nit:
Most times in this page, you should either write “Docker” or `docker`.

It's OK to skip that as this is moving existing content.

File: content/en/docs/reference/tools/map-crictl-dockercli.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+## Mapping from docker cli to crictl
+
+The exact versions for below mapping table are for docker cli v1.40 and crictl v1.19.0. Please note that the list is not exhaustive. For example, it doesn't include experimental commands of docker cli.
+
+{{< note >}}
+The output format of `crictl` is similar to Docker CLI, despite some missing columns for some CLI. Make sure to check output for the specific command if your command output is being parsed programmatically.
+{{< /note >}}
+
+### Retrieve Debugging Information
```
Comment (sftim) [inline content/en/docs/reference/tools/map-crictl-dockercli.md:26]: nit:
```suggestion
### Retrieve debugging information
```

It's OK to skip that as this is moving existing content.

**Review by sftim — COMMENTED** [review]
Only one thing that needs fixing; the shortcode `caution` is too strong for a deprecation notice. All else is just nits.

(Maybe a “deprecated” shortcode would help? There might be work under way to add one, I hear.)

File: content/en/docs/tasks/debug-application-cluster/crictl.md [hunk 10]
```diff
@@ -41,27 +37,36 @@ of Kubernetes. Extract it and move it to a location on your system path, such as
...
-runtimes, you can set the endpoint in multiple different ways:
+You can set the endpoint for `crictl` by doing one of the following:
 
-- By setting flags `--runtime-endpoint` and `--image-endpoint`
-- By setting environment variables `CONTAINER_RUNTIME_ENDPOINT` and `IMAGE_SERVICE_ENDPOINT`
-- By setting the endpoint in the config file `--config=/etc/crictl.yaml`
+* Set the `--runtime-endpoint` and `--image-endpoint` flags.
+* Set the `CONTAINER_RUNTIME_ENDPOINT` and `IMAGE_SERVICE_ENDPOINT` environment
+  variables.
+* Set the endpoint in the configuration file `--config=/etc/crictl.yaml`.
```
Comment (sftim) [inline content/en/docs/tasks/debug-application-cluster/crictl.md:45]: Is this also accurate:
```suggestion
* Set the endpoint in a configuration file and then specify the path to that file
  when you run `crictl`. For example: `crictl --config=/etc/crictl.yaml <subcommand>`.
```
?

File: content/en/docs/reference/tools/map-crictl-dockercli.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
+---
+title: Mapping from dockercli to crictl
+content_type: reference
+---
+
```
Comment (sftim) [inline content/en/docs/reference/tools/map-crictl-dockercli.md:5]: We might want to use the `{{% thirdparty-content %}}` shortcode here (I would).

File: content/en/docs/tasks/debug-application-cluster/crictl.md (hunk 10 above)
Comment (afbjorklund) [reply content/en/docs/tasks/debug-application-cluster/crictl.md:45]: Normally: Set the endpoint in the configuration file `/etc/crictl.yaml`. Unless you want to move it ?

**Comment by tengqm** [comment]
There are two issues here to be resolved:

1. Whether the crictl page belongs to the "debug-application-cluster". My answer would be a big no. We need to find a new home for this page.
2. There are some contents in the page that needs reorg.

The second depends on the first one. A related issue is about "kubectl" itself, which now resides in "tasks". That is not good. We may want to consolidate all these command line tools into a single subsection.

Before we have a clearer goal on this, I'm hesitant on any tweakings like this one.

**Comment by SergeyKanzhelev** [comment]
@shannonxtreme can you please only keep the section talking about the different socket that will be a default?

**Comment by shannonxtreme** [comment]
@tengqm this PR is purely for changes to make this doc ready for dockershim removal. I agree that we should talk about rehoming the crictl page and refining its content. I can open a separate issue to discuss that. 

@SergeyKanzhelev I'm not sure I understand - do you want me to remove the content about how to set the runtime endpoints etc, and only say that `crictl` tries to communicate with a list of known endpoints?

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev I'm not sure I understand - do you want me to remove the content about how to set the runtime endpoints etc, and only say that `crictl` tries to communicate with a list of known endpoints?

quite opposite. I don't care about re-homing, but changes regarding configuring endpoints is very useful

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Review by sftim — APPROVED** [review]
/approve
